interfaces/core.go

```diff
@@ -265,20 +265,37 @@ type StaticInfo struct {
 	AppArmorUnconfinedSlots bool
 }
 
+type PlugServiceSnippetSection string
+
+const (
+	PlugServiceSnippetUnitSection    PlugServiceSnippetSection = "unit"
+	PlugServiceSnippetServiceSection PlugServiceSnippetSection = "service"
+)
+
+// PlugServiceSnippet describes a systemd service snippet to be generated
+// for a snap with a plug whose interface implements ServicePermanentPlug.
+type PlugServiceSnippet struct {
+	// Section is the target unit file section for the snippet to be
+	// injected in (i.e. [Unit], [Service]).
+	Section PlugServiceSnippetSection
+	// Content is the actual snippet to be injected.
+	Content string
+}
```

Comment:
another approach would be do define an interface:

type PlugServicesSnippetSection interface {
   Section() string
   String() string
}

and two implementations:

type PlugServicesUnitSnippet  string

func (s PlugServicesUnitSnippet) Section() string {return "unit"}
func (s PlugServicesUnitSnippet) String() string {return string(s)}

etc

it might be more code here, but a little less code in the interfaces perhaps

Reply:
Thanks! updated. It is much cleaner in the interfaces side now.